chapter_optimization/lr-scheduler.md

```diff
@@ -63,6 +63,54 @@ def train(net, train_iter, test_iter, num_epochs, loss, trainer, device):
           f'test acc {test_acc:.3f}')
 ```
 
+```{.python .input}
+#@tab tensorflow
+%matplotlib inline
+from d2l import tensorflow as d2l
+import tensorflow as tf
+import math
+from tensorflow.keras.callbacks import LearningRateScheduler
+
+def net():
+    return tf.keras.models.Sequential([
+        tf.keras.layers.Conv2D(filters=6, kernel_size=5, activation='sigmoid',
+                               padding='same'),
+        tf.keras.layers.AvgPool2D(pool_size=2, strides=2),
+        tf.keras.layers.Conv2D(filters=16, kernel_size=5,
+                               activation='sigmoid'),
+        tf.keras.layers.AvgPool2D(pool_size=2, strides=2),
+        tf.keras.layers.Flatten(),
+        tf.keras.layers.Dense(120, activation='sigmoid'),
+        tf.keras.layers.Dense(84, activation='sigmoid'),
+        tf.keras.layers.Dense(10)])
+
+
+batch_size = 256
+train_iter, test_iter = d2l.load_data_fashion_mnist(batch_size=batch_size)
+
+# The code is almost identical to `d2l.train_ch6` that defined in the lenet
+# section of chapter convolutional neural networks
+def train(net_fn, train_iter, test_iter, num_epochs, lr,
+              device=d2l.try_gpu(), custom_callback = False):
+    """Train a model with a GPU (defined in Chapter 6)."""
+    device_name = device._device_name
+    strategy = tf.distribute.OneDeviceStrategy(device_name)
+    with strategy.scope():
+        optimizer = tf.keras.optimizers.SGD(learning_rate=lr)
+        loss = tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True)
+        net = net_fn()
+        net.compile(optimizer=optimizer, loss=loss, metrics=['accuracy'])
+    callback = d2l.TrainCallback(net, train_iter, test_iter, num_epochs,
+                             device_name)
+    if custom_callback is False:
+        net.fit(train_iter, epochs=num_epochs, verbose=0, 
+                callbacks=[callback])
+    else:
+         net.fit(train_iter, epochs=num_epochs, verbose=0,
+                 callbacks=[callback, custom_callback])
+    return net
```

Comment:
why is the trainer/optimizer and loss defined within the function?
If possible please follow MXNet style so that there is some sort of consistency.

Reply:
It's required, So that the model will be train in the GPU. We've to compile the model inside tf.distribute.Strategy scope.